src/Service/Models/GraphQLFilterParsers.cs

```diff
@@ -182,8 +181,21 @@ public GQLFilterParser(ISqlMetadataProvider metadataProvider)
                         }
                         else
                         {
+                            // This path will never get called for sql since the primary key will always required
+                            // This path will only be exercised for Cosmos
                             queryStructure.DatabaseObject.Name = sourceName + "." + backingColumnName;
                             queryStructure.SourceAlias = sourceName + "." + backingColumnName;
+                            string? nestedFieldType = _metadataProvider.GetSchemaGraphQLFieldTypeByFieldName(queryStructure.EntityName, name);
+
+                            if (nestedFieldType is null)
+                            {
+                                throw new DataApiBuilderException(
+                                    message: "Invalid filter object nested field input value type.",
```

Comment:
The message is a bit unclear. Are you trying to communicate that the field does not represent an object type on the GraphQL schema? i.e. if the field name were 'mars' but it is expected that the name is 'earth'? 

Does this convey your message?
> "Invalid filter object used as a nested field input value type."

Also, Wouldn't hotchocolate raise an exception when an incorrect field name is used in the request?

Reply:
Yes, hotchocolate does raise an exception already. So, techniquely, it shouldn't get here. I also updated the message to use what you suggested cause it's more clear.